fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanCharge.java

```diff
@@ -1101,4 +1105,18 @@ public ChargeTimeType getChargeTimeType() {
         return ChargeTimeType.fromInt(this.chargeTime);
     }
 
+    public LoanChargeData toData() {
+        EnumOptionData chargeTimeTypeData = new EnumOptionData((long) getChargeTimeType().ordinal(), getChargeTimeType().getCode(),
+                String.valueOf(getChargeTimeType().getValue()));
+        EnumOptionData chargeCalculationTypeData = new EnumOptionData((long) getChargeCalculation().ordinal(),
+                getChargeCalculation().getCode(), String.valueOf(getChargeCalculation().getValue()));
+        EnumOptionData chargePaymentModeData = new EnumOptionData((long) getChargePaymentMode().ordinal(), getChargePaymentMode().getCode(),
+                String.valueOf(getChargePaymentMode().getValue()));
+        Set<LoanInstallmentChargeData> loanInstallmentChargeDataSet = installmentCharges().stream().map(LoanInstallmentCharge::toData)
+                .collect(Collectors.toSet());
+        return new LoanChargeData(getId(), getCharge().getId(), getCharge().getName(), getCharge().toData().getCurrency(), amount,
```

Comment:
Probably a builder would be much better for readability in this case. (@Builder annotation from Lombok could be an option on the class)

Reply:
sure